### PR TITLE
EvaluateJavaScriptAsync with CommandMapper

### PR DESCRIPTION
**Pull request by hartez**
Implements the `EvaluateJavaScriptAsync` method for `WebView`; adds support for packing up arguments with a `Task` for communicating async results from a handler back to the cross-platform control. 

API changes:

ElementHandlerExtensions Added:

// Convenience method for invoking a command on a handler and getting back a result
`public static async Task<T> InvokeAsync<T>(this IElementHandler handler, string commandName, AsyncCommandArguments<T> args)`

// Same thing, but without the result
`public static async Task InvokeAsync(this IElementHandler handler, string commandName, AsyncCommandArguments args)`

Primitives Added:

// Contains a Task for returning a value of type T and letting the caller know the operation is finished
`public class AsyncCommandArguments<T>`

// Same thing as above, but no return value
`public class AsyncCommandArguments`

// Can pass in a script to execute and return the result as a string
`public class EvaluateJavaScriptAsyncRequest : AsyncCommandArguments<string>`

### CODE REVIEW

File: src/Core/src/Primitives/EvaluateJavaScriptAsyncRequest.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿using System;
+
+namespace Microsoft.Maui
+{
+	public record EvaluateJavaScriptAsyncRequest(string Script, Action<string> SetResult);
```
Comment (mattleibow) [inline src/Core/src/Primitives/EvaluateJavaScriptAsyncRequest.cs:5]: We need a way to propagate an exception, so maybe 2 Actions? or a combined `Action<string, Exception>`?

Comment (mattleibow) [reply src/Core/src/Primitives/EvaluateJavaScriptAsyncRequest.cs:5]: If we know that this is always an async, then we can even pass the `TaskCompletionSource` in here directly. The the callback just sets the right result/error.

File: src/Core/src/Primitives/EvaluateJavaScriptAsyncRequest.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	{
+		public string Script { get; }
+		
+		public EvaluateJavaScriptAsyncRequest(string script) : base()
+		{
+			Script = script;
+		}
+	}
+
+	public abstract class AsyncResultRequest<T>
```
Comment (mattleibow) [inline src/Core/src/Primitives/EvaluateJavaScriptAsyncRequest.cs:16]: ❤️

**Review by jsuarezruiz — COMMENTED** [review]
Need to update the WebViewStub in the tests:

`error CS0535: 'WebViewStub' does not implement interface member 'IWebView.EvaluateJavaScriptAsync(string)' [/Users/builder/azdo/_work/1/s/src/Core/tests/DeviceTests/Core.DeviceTests.csproj]`

**Comment by Redth** [comment]
Closing because #4507 is tastier.

**Pull request closed without merging**